### PR TITLE
Closes #2382 #2384 error messaging updates from admiraldev

### LINKED ISSUE #2382: General Issue: Fix errors for assert_list_of, assert_vars from admiraldev error messaging updates
### Background Information

image

### Definition of Done

Go!

### PR DESCRIPTION
**Pull request by bms63**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [ ] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [ ] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [ ] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [ ] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [ ] Review the [Cheat Sheet](https://github.com/pharmaverse/admiral/blob/main/inst/cheatsheet/admiral_cheatsheet.pdf). Make any required updates to it by editing the file `inst/cheatsheet/admiral_cheatsheet.pptx` and re-upload a PDF version of it to the same folder.
- [ ] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [ ] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [ ] Address any updates needed for vignettes and/or templates
- [ ] Update `NEWS.md` under the header `# admiral (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [ ] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [ ] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [ ] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [ ] Link the issue in the Development Section on the right hand side.
- [ ] Address all merge conflicts and resolve appropriately
- [ ] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: tests/testthat/test-derive_summary_records.R
Comment (bms63) [inline tests/testthat/test-derive_summary_records.R:1]: I'm unclear on why this was working with the "assert_s3_class", but in latest update it now falls over.  Removing it fixed it, but seems suspicious

Comment (ddsjoberg) [reply tests/testthat/test-derive_summary_records.R:1]: Hmmm, I just updated the description file to rely on the most recent admiraldev version, and it looks like everything is passing. Is there something else to check--am I missing something?

Comment (bms63) [reply tests/testthat/test-derive_summary_records.R:1]: That is the one I was checking locally with.  Also, the ci checks installs admiraldev from main - so they were previously failing.  The other PRs are failing from the updates

**Comment by bms63** [comment]
Anyways I don’t think k remotes is actually needed

**Comment by ddsjoberg** [comment]
> Anyways I don’t think k remotes is actually needed

Remotes isn't needed for the workflows that run on CI/CD. But one annoying that happens to users is they come across the admiral repo and install it from GitHub it won't work because you _need_ the dev version of admiraldev and without Remotes, there is no way for the installation process to know where to get it from.

File: tests/testthat/test-derive_summary_records.R
Comment (ddsjoberg) [reply tests/testthat/test-derive_summary_records.R:1]: Now that we've fixed issues here, is the plan to merge this PR and propagate the changes to the other open PRs? Or are we waiting to merge anything until all the `assert_*()` updates are complete?

**Comment by bms63** [comment]
> > Anyways I don’t think k remotes is actually needed
> 
> Remotes isn't needed for the workflows that run on CI/CD if that workflow uses staged deps to set up the packages. If a workflow isn't using staged deps, it will fail without Remotes. Also an annoying thing that happens to users is they come across the admiral repo and install it from GitHub it won't work because you _need_ the dev version of admiraldev and without Remotes, there is no way for the installation process to know where to get it from.
> 
> It's fine for us, the devs, to depend on tools like staged deps. But we should keep up our repo in a state so that anyone coming to the repo can install the package. This means adding Remotes when we depend on dev versions of packages.

Ah okay - I was thinking that our CI could use remotes now (thought that was a reason to rework CI R-CMD checks).

makes sense for users wishing to install dev versions

**Comment by bms63** [comment]
If everything looks okay can you approve so we can update the other PRs to pass CI

**Review by ddsjoberg — APPROVED** [review]
Looks great! Hopefully not too many more changes needed as we update admiraldev 🤞🏼

**Pull request merged**